File: 509_fibbonacci/sol.py
class Solution:
    def fib(self, n:int) -> int:
        fibNum = 0
        if n == 1:
            return 1
        elif n == 2:
            return 1
        elif n >= 3:
            fibo = [0] * n
            fibo[0] = 1
            fibo[1] = 1
            for i in range(2, n):
                fibo[i] = fibo[i-1] + fibo[i-2]
            fibNum = fibo[n-1]
        return fibNum

p = Solution()
print(p.fib(6))

# link https://leetcode.com/problems/fibonacci-number/

File: 509_fibbonacci/sol.cpp
class Solution {
public:
    int fib(int n) {
        int fibNum = 0;
        if(n < 2) return n;
        int fibo[30] = {};
        fibo[0] = 1;
        fibo[1] = 1;
        for(int i=2; i<n; i++){
            fibo[i] = fibo[i-1] + fibo[i-2];
        }
        fibNum = fibo[n-1]; 
        return fibNum;
        
    }
};